src/App.js

```diff
@@ -3,20 +3,35 @@ import Home from './pages/Home';
 import { Link, Route, Routes } from 'react-router-dom';
 import Products from './pages/Products';
 import Product from './pages/Product';
-//import { Navbar, NavbarHeader, NavbarItems, NavItem } from 'react-bootstrap-navbar';
+import { ThemeContext, themes } from './context/themeContext';
+import { useState } from 'react';
+import ToggleSwitch from './components/ToggleSwitch';
 
 function App() {
+  const [theme, setTheme] = useState(themes.dark);
```

Comment:
The entire background does not change with the theme. After loading the products list page, scroll to the bottom to see white, you can also tell you're not setting the background correctly when you over scroll in dark mode. The fix required a little digging on your end but this is how you'd do it.

```
document.body.style.background = theme.background;
```
If you end up refactoring based on the links I posted, then you'd want this piece of code within the useEffect of your provider.